rakelib/artifacts.rake

```diff
@@ -94,13 +94,25 @@ namespace "artifact" do
   end
 
   task "prepare" => ["bootstrap", "plugin:install-default", "install-logstash-core", "install-logstash-core-event", "clean-bundle-config"]
+  task "prepare-all" => ["bootstrap", "plugin:install-all", "install-logstash-core", "install-logstash-core-event", "clean-bundle-config"]
 
-  desc "Build a tar.gz of logstash with all dependencies"
+  desc "Build a tar.gz of default logstash plugins with all dependencies"
   task "tar" => ["prepare"] do
+    puts("[artifact:tar] Building tar.gz of default plugins")
+    build_tar
+  end
+
+  desc "Build a tar.gz of all logstash plugins from logstash-plugins github repo"
+  task "tar-all-plugins" => ["prepare-all"] do
+    puts("[artifact:tar] Building tar.gz of all plugins")
+    build_tar "all-plugins"
+  end
+
+  def build_tar(tar_suffix = "")
     require "zlib"
     require "archive/tar/minitar"
     require "logstash/version"
-    tarpath = "build/logstash-#{LOGSTASH_VERSION}.tar.gz"
+    tarpath = "build/logstash-#{tar_suffix}-#{LOGSTASH_VERSION}.tar.gz"
```

Comment:
This will not be backward compatible with the previous naming convention since if no suffix is provided we will have two `-` in the name. I think the easy fix is to set `NIL` as the default value of `tar_suffix` and then use this method.

 tarpath = ["build/logstash", tar_suffix, "{LOGSTASH_VERSION}.tar.gz"].compact.join("-")